test/spec/common/client/set-cache.js

```diff
@@ -0,0 +1,48 @@
+  var searchTerm = 'search';
+  var cacheKey = '/1/indexes/' + credentials.indexName + '/query_body_{"params":"query=' + searchTerm + '"}';
+  var cache = {};
+  cache[cacheKey] = cachedResult;
+
+  var opts = {
+    cache: cache
+  };
+
+  var algoliasearch = require('../../../../');
+  var searchWithCache = algoliasearch(credentials.applicationID, credentials.searchOnlyAPIKey, opts);
+
+  var indexWithCache = searchWithCache.initIndex(credentials.indexName);
+
+  fauxJax.on('request', function(req) {
+    req.respond(200, {}, cachedResult);
```

Comment:
You should not even have to use fauxJax, since if there's a request it means that your "prefill" cache feature failed.

What I would do is put a t.fail() inside the .on('request') instead.